File: MyWatchlist/MyWatchlistSpring/src/main/java/it/niko/mywatchlist/payload/request/WatchlistRequest.java
package it.niko.mywatchlist.payload.request;

import lombok.Getter;
import lombok.Setter;
import javax.validation.constraints.NotBlank;
import javax.validation.constraints.Size;

@Getter
@Setter
public class WatchlistRequest {
    @NotBlank
    @Size(max = 50)
    private String title;

    @NotBlank
    @Size(max = 20)
    private String username;

    private String status;

    private int progress;

    private int score;

    @Size(max = 200)
    private String comment;
}

File: MyWatchlist/MyWatchlistAndroid/app/src/main/java/it/niko/mywatchlistandroid/payload/WatchlistRequest.kt
package it.niko.mywatchlistandroid.payload

import com.google.gson.annotations.SerializedName

data class WatchlistRequest(
    @SerializedName("title")
    val title: String,

    @SerializedName("username")
    val username: String,

    @SerializedName("status")
    val status: String,

    @SerializedName("progress")
    val progress: Int,

    @SerializedName("score")
    val score: Int,

    @SerializedName("comment")
    val comment: String,
)
